### PR TITLE
Add gnucobol

### PR DESCRIPTION
**Pull request by pavelzw**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by pavelzw** [comment]
@conda-forge/help-c-cpp, this is ready for review

File: recipes/gnucobol/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  run:
+    - gmp
+    - json-c
+    - libxml2
+    - ncurses
+    - libdb
+
+test:
+  commands:
+    - cobc --version
```
Comment (xhochy) [inline recipes/gnucobol/meta.yaml:37]: As this is a compiler, could you also add a test that it produces working binaries?

File: recipes/gnucobol/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    - json-c
+    - libxml2
+    - ncurses
+    - libdb
+  run:
+    - gmp
+    - json-c
+    - libxml2
+    - ncurses
+    - libdb
```
Comment (xhochy) [inline recipes/gnucobol/meta.yaml:33]: These should all come in automatically via their `run_exports`.
```suggestion
```

File: recipes/gnucobol/build.sh [hunk 3]
```diff
@@ -0,0 +1,13 @@
+#!/usr/bin/env bash
+
```
Comment (xhochy) [inline recipes/gnucobol/build.sh:2]: Run this in verbose strict mode please.
```suggestion
set -exuo pipefail
```

File: recipes/gnucobol/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    - {{ compiler('c') }}
+    - {{ stdlib('c') }}
+    - pkg-config
+    - make
+  host:
+    - gmp
+    - json-c
+    - libxml2
+    - ncurses
+    - libdb
```
Comment (xhochy) [inline recipes/gnucobol/meta.yaml:27]: This needs libintl on non-Linux platforms.
```suggestion
    - libdb
    - libintl-devel  # [not linux]
```

See in the `osx-64` build:
```
2024-05-03T13:07:27.6609120Z checking for GNU gettext in libintl... no
```

File: recipes/gnucobol/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+build:
+  number: 0
+  skip: True  # [win]
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ stdlib('c') }}
+    - pkg-config
+    - make
```
Comment (xhochy) [inline recipes/gnucobol/meta.yaml:21]:
```suggestion
    - make
    - gettext-tools
```

**Review by xhochy — CHANGES_REQUESTED** [review]
On `osx-64` you get

```
2024-05-03T13:08:47.2821510Z WARNING (gnucobol): run-exports library package conda-forge/osx-64::libdb==6.2.32=he49afe7_0 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
2024-05-03T13:08:47.2823420Z WARNING (gnucobol): run-exports library package conda-forge/osx-64::libxml2==2.12.6=hcacace5_2 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
```

Are these features maybe not correctly detected?

**Review by xhochy — APPROVED** [review]

**Pull request merged**